core/src/main/java/dk/alexandra/fresco/lib/collections/sort/OddEvenMergeProtocolImpl.java

```diff
@@ -26,133 +26,140 @@
 import dk.alexandra.fresco.framework.ProtocolProducer;
+import dk.alexandra.fresco.framework.builder.ComputationBuilder;
+import dk.alexandra.fresco.framework.builder.binary.ProtocolBuilderBinary;
 import dk.alexandra.fresco.framework.util.Pair;
 import dk.alexandra.fresco.framework.value.SBool;
-import dk.alexandra.fresco.lib.helper.SimpleProtocolProducer;
-import dk.alexandra.fresco.lib.helper.builder.BasicLogicBuilder;
-import dk.alexandra.fresco.lib.logic.AbstractBinaryFactory;
 import java.util.List;
 
 /**
  * An implementation of the OddEvenMergeProtocol. This does not support threading,
  * the OddEvenMergeProtocolRec class should be preferable to this.
- * 
+ *
  * @author psn
  *
  */
-public class OddEvenMergeProtocolImpl extends SimpleProtocolProducer implements
-    OddEvenMergeProtocol {
+public class OddEvenMergeProtocolImpl implements
+    ComputationBuilder<List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>>, ProtocolBuilderBinary> {
 
-	private BasicLogicBuilder blb;
-	private List<Pair<SBool[], SBool[]>> sorted;
-	private List<Pair<SBool[], SBool[]>> left;
-	private List<Pair<SBool[], SBool[]>> right;
-	private int firstIndex;
-	private int lastIndex;
-	private int realSize;
-	private int simulatedSize;
+  private List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>> left;
+  private List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>> right;
 
-	public OddEvenMergeProtocolImpl(List<Pair<SBool[], SBool[]>> left,
-			List<Pair<SBool[], SBool[]>> right,
-			List<Pair<SBool[], SBool[]>> sorted, AbstractBinaryFactory factory) {
-		super();
-		this.blb = new BasicLogicBuilder(factory);
-		this.sorted = sorted;
-		this.left = left;
-		this.right = right;
-	}
+  public OddEvenMergeProtocolImpl(
+      List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>> left,
+      List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>> right) {
+    super();
+    this.left = left;
+    this.right = right;
+  }
 
-	@Override
-	protected ProtocolProducer initializeProtocolProducer() {
-		
-		blb.beginParScope();
-		for (int i = 0; i < left.size(); i++) {
-			Pair<SBool[], SBool[]> leftPair = left.get(i);
-			Pair<SBool[], SBool[]> upperPair = sorted.get(i);
-			blb.copy(leftPair.getFirst(), upperPair.getFirst());
-			blb.copy(leftPair.getSecond(), upperPair.getSecond());
-		}
-		for (int i = 0; i < right.size(); i++) {
-			Pair<SBool[], SBool[]> rightPair = right.get(i);
-			Pair<SBool[], SBool[]> lowerPair = sorted.get(i + left.size());
-			blb.copy(rightPair.getFirst(), lowerPair.getFirst());
-			blb.copy(rightPair.getSecond(), lowerPair.getSecond());
-		}
-		blb.endCurScope();
-		initializeIndices();
-		newMerge(0, simulatedSize, 1);
-		return blb.getProtocol();
-	}
+  @Override
+  public Computation<List<Pair<List<Computation<SBool>>, List<Computation<SBool>>>>> build(
+      ProtocolBuilderBinary builder) {
+    // TODO Auto-generated method stub
+    return null;
```

Comment:
This seems unhealthy - could we complete the refactor here (or remove it all together)